src/cmd_handlers.c

```diff
@@ -1035,20 +1035,25 @@ regedit_cmd(const cmd_info_t *cmd_info)
 	}
 	char tmp_fname[PATH_MAX + 1];
 	generate_tmp_file_name("vifm.regedit", tmp_fname, sizeof(tmp_fname));
+
+	mode_t saved_umask = umask(~0600);
 	write_file_of_lines(tmp_fname, reg->files, reg->nfiles);
 	// Non-zero result means there's an error occured.
 	const int edit_result = vim_view_file(tmp_fname, 1, 1, /*allow_forking=*/1);
-	// TODO: handle editing error
-
-	regs_clear(regname);
-	reg->files = read_file_of_lines(tmp_fname, &reg->nfiles);
+	(void)umask(saved_umask);
 	// TODO: implement temporary file deletion
+	// TODO: handle editing error
 
-	if(reg->files == NULL)
+	int read_lines = 0;
+	strlist_t *edited_content = read_file_of_lines(tmp_fname, &read_lines);
+	if (edited_content == NULL)
 	{
-		ui_sb_err("Failed to write editions to register.");
+		// TODO: read errno and modify message according to it.
```

Comment:
No need for this.  `errno` isn't reliable at this point because `read_file_of_lines()` doesn't try to preserve it.

Reply:
My bad, I missed that `read_file_of_lines()` promises to preserve `errno`.